### ISSUE TITLE
[Retail] Group loot question

### ISSUE THREAD

**Issue by TreizeEU** [Xuerian/XLoot#19]
Are the Group loot module still broken/unusable (since DF pre-patch)?

I ran LFR last night, and got the below error. Group loot frame pops up when a boss is looted, but it has no text and seems to stay there until I reload my UI (even if the roll timer runs out).

`311x FontString:SetText(): Font not set
[string "=[C]"]: in function `SetText'
[string "@XLoot_Group/Group.lua"]:785: in function `SetText'
[string "@XLoot_Group/Group.lua"]:451: in function `?'
[string "@XLoot/XLoot.lua"]:39: in function <XLoot/XLoot.lua:37>

Locals:
(*temporary) = FontString {
 0 = <userdata>
}
(*temporary) = "1"
`

**Comment by Xuerian** [comment]
I can't reproduce this error.

Are you getting any other errors?

Have you tried reinstalling the mod?

What happens when you use /xlgd ?

**Comment by TreizeEU** [comment]
I think that's the only error I'm getting, and /xlgd does nothing for me? I only get a chat-error saying "Type '/help' for a listing of a few commands"?

Could it be caused by the font I use?

I will run LFR again today so I can test font-settings between bosses.

**Comment by Xuerian** [comment]
It's possible, but I would imagine there would be another error first. It sets the font when it initializes, and that happens before the error you're giving me. 

You can try installing the unpackaged version from here, remove/move out what you have and unpack https://github.com/Xuerian/XLoot/archive/refs/heads/master.zip to Interface/Addons (you may need to rename the main folder to "XLoot", I can't recall if wow cares), so you have Addons/XLoot/XLoot.lua etc.

Then you can use /xlgd to test group wherever.

**Comment by TreizeEU** [comment]
Update: I killed a LFR boss and no rolling frame popped up at all (I had used the /xlgd command before the fight).

Not sure if these are the same, but I triggered two more lua errors from that session:

`4x XLoot/Modules/Group/Group.lua:1141: bad argument #1 to 'unpack' (table expected, got nil)
[string "=[C]"]: in function `unpack'
[string "@XLoot/Modules/Group/Group.lua"]:1141: in function <XLoot/Modules/Group/Group.lua:1140>
[string "@XLoot/Modules/Group/Group.lua"]:234: in function `?'
[string "@XLoot/XLoot.lua"]:39: in function <XLoot/XLoot.lua:37>

Locals:
(*temporary) = nil
(*temporary) = "table expected, got nil"`

And:

`89x XLoot/Modules/Group/Group.lua:1168: attempt to index field '?' (a nil value)
[string "@XLoot/Modules/Group/Group.lua"]:1168: in function <XLoot/Modules/Group/Group.lua:1167>
[string "@XLoot/Modules/Group/Group.lua"]:387: in function `?'
[string "@XLoot/XLoot.lua"]:39: in function <XLoot/XLoot.lua:37>

Locals:
hid = 3
(*temporary) = <function> defined =[C]:-1
(*temporary) = nil
(*temporary) = "attempt to index field '?' (a nil value)"
FakeHistory = <table> {
 items = <table> {
 }
 rolls = <table> {
 }
 links = <table> {
 }
}`

**Comment by Xuerian** [comment]
Please test with only XLoot enabled - You may need to install and enable the dependencies, Ace3, separately. 

/xlgd is "xloot group debug", and the goal is to use while just standing around out of a group to show a group loot frame that mostly tests things. Itwill definitely break group if it wasn't already broken until you /reloadui as it swaps in testing functions.

Sorry about not explaining that thoroughly.

Please let me know what happens when you try that.

**Comment by TreizeEU** [comment]
So, I did that and the fake item rolling frame showed up, seemed to disappear like it should once the timer ran out. 

This did cause a bunch new lua errors though (even if I download and enable Ace3 as a separate addon).

[!BugGrabber.zip](https://github.com/Xuerian/XLoot/files/10789706/BugGrabber.zip)

**Comment by Xuerian** [comment]
Any of the "Error loading/Couldn't open" errors are fine. They're a side effect of running the addon without the libraries inside of it.

If you get any errors that aren't those two things, you should post them here.

You can try enabling your other addons and seeing what /xlgd does, too.

**Comment by TreizeEU** [comment]
Using that command with the regular set of addons enabled, causes a chat message "XLoot Group: Ignoring STAR_LOOT_ROLL" with no name", it doesn't cause a fake loot roll.

Fresh BugGrabber.lua file after deleting the old file and enabled addons:

[!BugGrabber.zip](https://github.com/Xuerian/XLoot/files/10790115/BugGrabber.zip)

**Comment by Xuerian** [comment]
"Module 'Options' already exists" implies you have the separate modules enabled at the same time as "XLoot" (Combined package). Disable "XLoot Frame" etc.

**Comment by Xuerian** [comment]
Yes, the couldn't load errors are fine, ignore them. If they bother you, you can comment them out in the .toc files, but it doesn't matter. I'm just looking for any errors NOT involving them.

You can leave LibStub and CallbackHandler installed as separate addons if you had them. Only need the "XLoot {module}" folders out.

So /xlgd works with that. Do they work with real loot? (Remember to /reloadui if you used /xlgd)

**Comment by TreizeEU** [comment]
So, I just did another run of LFR and the group module seems to work normally now. Do I need to live with those lua errors, then, or is there a way to fix them other than commenting out lines in the xloot.toc file after every update (presumably)?

**Comment by Xuerian** [comment]
If the git version (this one) works, the live version should also work. They're the same code. Try just cleanly reinstalling it normally. (Remove or disable "XLoot")

**Comment by TreizeEU** [comment]
So I delete the XLoot folder in Interface/Addons, then reinstall it through the curseforge client?

**Comment by Xuerian** [comment]
Yes. Delete any XLoot folder inside of Interface/Addons.

**Comment by TreizeEU** [comment]
Done, I'm guessing it's fine that it reinstalled the modules (frames, monitor, etc) again? At least I'm not getting those couldn't load errors now, so that's good? :)

I will probably run another round of LFR tomorrow after reset to see if all's good with the group module.

I really don't know what was broken, but thanks for helping sort this out.

**Comment by Xuerian** [comment]
It should, yes. The Curseforge packager takes the zip you downloaded, removes debugging stuff, patches in localizations, and extracts the modules (frame/etc) into their own folders.

Let me know how it goes. Glad to help.

**Comment by TreizeEU** [comment]
Yup, the group module seems to work now, no lua errors after running the first wing of LFR. You can probably close this ticket now. :)

**Comment by Xuerian** [comment]
Sounds good.

Sometimes turning it off and on again with malicious intent really is the answer.
